src/text/TextComponent.cpp

```diff
@@ -0,0 +1,261 @@
+#include <format>
+#include <iostream>
+
+#include <scene/components/MeshComponent.h>
+#include <scene/components/TransformComponent.h>
+#include <text/TextComponent.h>
+
+namespace arch::text {
+
+TextComponent::TextComponent(ecs::Domain& domain, const ecs::Entity entity) noexcept:
+	_domain{ &domain },
+	_entity{ entity } {
+	_domain
+		->addComponent<scene::components::TransformComponent>(_entity, float3{}, Quat{ 0, 0, 0, 1 }, float3{ 1, 1, 0 });
+}
+
+TextComponent& TextComponent::operator=(font::Face& face) noexcept {
+	return setFont(face);
+}
+
+TextComponent& TextComponent::setFont(std::string_view familyName, std::string_view styleName) {
+	auto&& fontDB = font::FontDB::get();
+
+	auto fontOpt = fontDB[familyName];
+	if (!fontOpt) {
+		throw font::FontNotFoundException(std::format("Font '{}' not found! Try FontDB::exists", familyName));
+	}
+
+	auto faceOpt = fontOpt.get()[styleName];
+	if (!fontOpt) {
+		throw font::FontNotFoundException(
+			std::format("Style '{}' not found in font '{}'! Try Font::hasStyle", styleName, familyName)
+		);
+	}
+
+	_face = &*faceOpt;
+	_face->assure();
+
+	return *this;
+}
+
+TextComponent& TextComponent::setFont(font::Face& face) {
+	_face = &face;
+	_face->assure();
+	return *this;
+}
+
+TextComponent& TextComponent::setPosition(float2 pos) {
+	auto&& t = _domain->addComponent<scene::components::TransformComponent>(_entity);
```

Comment:
Component shouldn't modify other components.
ECS Component should be just container for data, nothing more.